diff --git a/ios/Legacy/Services/EXReactNativeAdapter.mm b/ios/Legacy/Services/EXReactNativeAdapter.mm
index b1d0f5988d61b7346500ac65fcb7714903ce4398..b83df54a2133be5e73e4912148452beb62dec0a7 100644
--- a/ios/Legacy/Services/EXReactNativeAdapter.mm
+++ b/ios/Legacy/Services/EXReactNativeAdapter.mm
@@ -35,7 +35,7 @@ + (NSString *)moduleName
 
 + (const NSArray<Protocol *> *)exportedInterfaces
 {
-  return @[@protocol(EXAppLifecycleService), @protocol(EXUIManager), @protocol(EXJavaScriptContextProvider)];
+  return (const NSArray<Protocol *> * _Nonnull)@[@protocol(EXAppLifecycleService), @protocol(EXUIManager), @protocol(EXJavaScriptContextProvider)];
 }
 
 # pragma mark - Lifecycle methods
