diff --git a/ios/ExpoAsset.podspec b/ios/ExpoAsset.podspec
index 832decbfd3de2b5c11f70d10ece92d7be69a409b..5e313fa3550c84c024ff4b603bc7b0d652bd9ab5 100644
--- a/ios/ExpoAsset.podspec
+++ b/ios/ExpoAsset.podspec
@@ -12,7 +12,8 @@ Pod::Spec.new do |s|
   s.homepage       = package['homepage']
   s.platforms      = {
     :ios => '15.1',
-    :tvos => '15.1'
+    :tvos => '15.1',
+    :osx => '12.0'
   }
   s.swift_version  = '5.4'
   s.source         = { git: 'https://github.com/expo/expo.git' }
