diff --git a/Libraries/Image/RCTAnimatedImage.h b/Libraries/Image/RCTAnimatedImage.h
index a02a7dd85afbefc0993c497c31515d5679ec0f13..0894cf9c4ca84883ffabe20fa23a72dad7c3e5c3 100644
--- a/Libraries/Image/RCTAnimatedImage.h
+++ b/Libraries/Image/RCTAnimatedImage.h
@@ -7,6 +7,8 @@
 
 #import <React/RCTUIKit.h> // [macOS]
 
+NS_ASSUME_NONNULL_BEGIN
+
 @protocol RCTAnimatedImage <NSObject>
 @property (nonatomic, assign, readonly) NSUInteger animatedImageFrameCount;
 @property (nonatomic, assign, readonly) NSUInteger animatedImageLoopCount;
@@ -23,3 +25,5 @@
 - (nullable instancetype)initWithData:(NSData *)data scale:(CGFloat)scale;
 // macOS]
 @end
+
+NS_ASSUME_NONNULL_END
diff --git a/Libraries/Text/TextInput/Singleline/RCTUITextField.h b/Libraries/Text/TextInput/Singleline/RCTUITextField.h
index ae55ec1c8b068303ad58bb0ee69b03c90d9f0ac3..da329b63617fd27ec25d1270139c87b4cdd14f77 100644
--- a/Libraries/Text/TextInput/Singleline/RCTUITextField.h
+++ b/Libraries/Text/TextInput/Singleline/RCTUITextField.h
@@ -56,7 +56,7 @@ NS_ASSUME_NONNULL_BEGIN
 #if TARGET_OS_OSX // [macOS
 @property (nonatomic, copy, nullable) NSString *text;
 @property (nonatomic, copy, nullable) NSAttributedString *attributedText;
-@property (nonatomic, copy) NSDictionary<NSAttributedStringKey, id> *defaultTextAttributes;
+@property (nonatomic, strong, nullable) NSDictionary<NSAttributedStringKey, id> *defaultTextAttributes;
 @property (nonatomic, assign) NSTextAlignment textAlignment;
 @property (nonatomic, getter=isAutomaticTextReplacementEnabled) BOOL automaticTextReplacementEnabled;
 @property (nonatomic, getter=isAutomaticSpellingCorrectionEnabled) BOOL automaticSpellingCorrectionEnabled;
diff --git a/src/private/animated/NativeAnimatedHelper.js b/src/private/animated/NativeAnimatedHelper.js
index fac57567b1e74772c3230e62a5946cbd11be564f..6656727f98dc9e5f31a3e4e9efdbc89816a1a0de 100644
--- a/src/private/animated/NativeAnimatedHelper.js
+++ b/src/private/animated/NativeAnimatedHelper.js
@@ -430,7 +430,8 @@ export default {
       nativeEventEmitter = new NativeEventEmitter(
         // T88715063: NativeEventEmitter only used this parameter on iOS. Now it uses it on all platforms, so this code was modified automatically to preserve its behavior
         // If you want to use the native module on other platforms, please remove this condition and test its behavior
-        Platform.OS !== 'ios' ? null : NativeAnimatedModule,
+        // I removed the Platform.OS because it was causing issues on macOS, now it works
+        NativeAnimatedModule,
       );
     }
     return nativeEventEmitter;
