diff --git a/node_modules/react-native-macos/.bun-tag-8977212d81cbb5a b/.bun-tag-8977212d81cbb5a
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/scripts/cocoapods/helpers.rb b/scripts/cocoapods/helpers.rb
index 61a5575f7c24132e8a1020ff9bbf298f15d0a0c9..16b7c931188b691ed791202e28ba9f3e4922a186 100644
--- a/scripts/cocoapods/helpers.rb
+++ b/scripts/cocoapods/helpers.rb
@@ -85,7 +85,7 @@ module Helpers
         end
         # [macOS
         def self.min_macos_version_supported
-            return '14.0'
+            return '11.0'
         end
         # macOS]
         # [visionOS
diff --git a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
index fdd7e95c3f91cb35cd58d86315456e61df97ffd2..5b4ef6b2e04ef70b5a601a363995609cf62b25fc 100644
--- a/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
+++ b/scripts/codegen/generate-artifacts-executor/generateRCTThirdPartyComponents.js
@@ -138,14 +138,14 @@ function generateRCTThirdPartyComponents(
     .flatMap(library => {
       const components = componentsInLibraries[library];
       return components.map(({componentName, className}) => {
-        return `\t\t@"${componentName}": NSClassFromString(@"${className}"), // ${library}`;
+        return `\t\t\tClass ${className}ObjC = NSClassFromString(@"${className}");\n\t\t\tif (${className}ObjC) [components setObject:${className}ObjC forKey:@"${componentName}"];\n\n`;
       });
     })
     .join('\n');
   // Generate implementation file
   const templateMM = fs
     .readFileSync(THIRD_PARTY_COMPONENTS_MM_TEMPLATE_PATH, 'utf8')
-    .replace(/{thirdPartyComponentsMapping}/, thirdPartyComponentsMapping);
+    .replace(/{thirdPartyComponentsMappingWithSafteyChecks}/, thirdPartyComponentsMapping);
   const finalPathMM = path.join(
     outputDir,
     'RCTThirdPartyComponentsProvider.mm',
diff --git a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
index 4ce510709e337ce783a52413063b2c1ac6fcb29a..47d9fd24e75b605daf0c379625a0fa2ab1d4a4f0 100644
--- a/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
+++ b/scripts/codegen/templates/RCTThirdPartyComponentsProviderMM.template
@@ -19,9 +19,9 @@
   static dispatch_once_t nativeComponentsToken;
 
   dispatch_once(&nativeComponentsToken, ^{
-    thirdPartyComponents = @{
-{thirdPartyComponentsMapping}
-    };
+    NSMutableDictionary<NSString *, Class<RCTComponentViewProtocol>> *components = [NSMutableDictionary dictionary];
+    {thirdPartyComponentsMappingWithSafteyChecks}
+    thirdPartyComponents = [components copy];
   });
 
   return thirdPartyComponents;
